virttest/qemu_monitor.py

```diff
@@ -3635,3 +3636,46 @@ def query_sgx_capabilities(self):
         cmd = "query-sgx-capabilities"
         self.verify_supported_cmd(cmd)
         return self.cmd(cmd)
+
+    def query_sev(self):
+        """
+        Query SEV information. (Since 2.12)
+
+        :return: Information about SEV
+        """
+        cmd = "query-sev"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_capabilities(self):
+        """
+        Query SEV capabilities. (Since 2.12)
+
+        :return: SEV capability objects.
+        """
+        cmd = "query-sev-capabilities"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_launch_measure(self):
+        """
+        Query the SEV guest launch information. (Since 2.12)
+
+        :return: The SEV launch measure info for the guest.
+        """
+        cmd = "query-sev-launch-measure"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_attestation_report(self):
+        """
+        Query SEV attestation report. (Since 6.1)
+        Note that it needs to generate a random 16 bytes value encoded
+        in base64 to query attestation report.
+
+        :return: The SEV attestation report objects.
+        """
+        cmd = "query-sev-attestation-report"
+        mnonce = base64.b64encode(os.urandom(16)).decode()
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd, {"arguments": {"mnonce": mnonce}})
```

Comment:
```suggestion
    def query_sev_attestation_report(self, mnonce):
        """
        Query SEV attestation report. (Since 6.1)
 
        :param mnonce: A random 16 bytes value encoded in base64
        :type mnonce: str
        :return: The SEV attestation report objects.
        """
        cmd = "query-sev-attestation-report"
        self.verify_supported_cmd(cmd)
        return self.cmd(cmd, {"arguments": {"mnonce": mnonce}})
```
Here, we focus on warping the qmp command `query-sev-attestation-report` according to https://github.com/qemu/qemu/blob/c152379422a204109f34ca2b43ecc538c7d738ae/qapi/misc-target.json#L215, not better to include the upper-level logic, we could let the caller handle the generator of the mnonce and then pass it to the function.
BTW, we could not provide the default value for it, beside the qmp will generate a default value if we did not pass the `mnonce` argument, like https://github.com/qemu/qemu/blob/c152379422a204109f34ca2b43ecc538c7d738ae/qapi/block.json#L328-L334 and https://github.com/avocado-framework/avocado-vt/blob/7c8692a2e2fa21307c742af549f082bb2b4c07b1/virttest/qemu_monitor.py#L2842-L2871